### PR TITLE
CLN: Exception catching in expressions

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/computation/expressions.py [hunk 1]
```diff
@@ -107,15 +107,12 @@ def _evaluate_numexpr(op, op_str, a, b, reversed=False):
...
         b_value = getattr(b, "values", b)
-        try:
-            result = ne.evaluate(
-                "a_value {op} b_value".format(op=op_str),
-                local_dict={"a_value": a_value, "b_value": b_value},
-                casting="safe",
-            )
-        except ValueError as detail:
-            if "unknown type object" in str(detail):
-                pass
```
Comment (jorisvandenbossche) [inline pandas/core/computation/expressions.py]: This did never happen?

File: pandas/core/computation/expressions.py [hunk 2]
```diff
@@ -167,11 +158,10 @@ def _where_numexpr(cond, a, b):
...
 
 def _has_bool_dtype(x):
+    if isinstance(x, ABCDataFrame):
+        return "bool" in x.dtypes
     try:
-        if isinstance(x, ABCDataFrame):
-            return "bool" in x.dtypes
-        else:
-            return x.dtype == bool
+        return x.dtype == bool
```
Comment (jreback) [inline pandas/core/computation/expressions.py]: you can use is_bool_dtype(x) ?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/computation/expressions.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/computation/expressions.py]: could, but thats non-performant, which is a big deal here

File: pandas/core/computation/expressions.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/computation/expressions.py]: the only errors that bubble up are TypeErrors, which we catch in the calling func in ops.array_ops

File: pandas/core/computation/expressions.py (hunk 2 above)
Comment (jreback) [reply pandas/core/computation/expressions.py]: k

**Review by jreback — APPROVED** [review]

**Pull request merged**